File: src/main/java/kr/co/wanted/market/common/global/dto/ErrorResponse.java
package kr.co.wanted.market.common.global.dto;

import lombok.Getter;

import java.util.Map;

@Getter
public class ErrorResponse {

    private final String code;

    private final String message;

    private final Map<String, String> basicValidations;


    private ErrorResponse(String code,
                          String message,
                          Map<String, String> basicValidations) {

        this.code = code;
        this.message = message;
        this.basicValidations = basicValidations;
    }


    public static ErrorResponse of(String code,
                                   String message,
                                   Map<String, String> basicValidations) {

        return new ErrorResponse(code, message, basicValidations);
    }

}

File: src/main/kotlin/wanted/market/api/common/security/ErrorResponse.kt
package wanted.market.api.common.security

import wanted.market.api.common.ErrorCode
import java.time.LocalDateTime

class ErrorResponse(
    var message: String,
    var code: String,
    var timestamp: String = LocalDateTime.now().toString()
) {

    companion object {
        fun of(errorCode: ErrorCode): ErrorResponse {
            return ErrorResponse(errorCode.message, errorCode.code)
        }

        fun of(errorMessage: String): ErrorResponse {
            return ErrorResponse(errorMessage, "400")
        }
    }
}
